operator/controllers/resources/role_controller_test.go

```diff
@@ -73,6 +75,62 @@ func TestRoleCreation(t *testing.T) {
 	})
 }
 
+func TestRoleCreationFromYAML(t *testing.T) {
+	ctx := context.Background()
+
+	teleportServer, operatorName := defaultTeleportServiceConfig(t)
+
+	require.NoError(t, teleportServer.Start())
+
+	tClient := clientForTeleport(t, teleportServer, operatorName)
+	k8sClient := startKubernetesOperator(t, tClient)
+
+	namespace := createNamespaceForTest(t, k8sClient)
+	roleName := "interns"
+
+	roleYAML := yaml.NewYAMLOrJSONDecoder(bytes.NewBuffer([]byte(`
```

Comment:
Do you mind refactoring into a table test, so we can add more examples? (Like the totally absurd one I mentioned earlier?)